Graphics for most workers and some immovables

Noordfrees opened the pull request:
Completed
- all workers except soldier and tapir
- ferry and ship
- shipconstruction
- resis
- rare trees

Noordfrees committed:
Completed a batch of workers:
- carrier
- charcoal burner
- cook
- dressmaker
- gold digger
- gold smelter
- scout
- stonecarver
- tapir breeder
- trainer

actions-user committed:
92 files were automatically formatted.

gunchleoc commented:
I have finally given the Amazons a very quick test. I think the flags and border markers as well as the carriers' dresses should be brighter. Maybe get some yellow in there?

Noordfrees committed:
Adjust carrier

Noordfrees committed:
Adjusted flag and frontier visibility

Noordfrees committed:
Merged master

Noordfrees commented:
Changed the carrier dress from midnight blue to violet. I edited the flags and frontiers in GIMP, the *Belichtung*-Tool is very useful here. What do you think?

gunchleoc commented:
That's better - the frontier markers now look as if they are glowing though. It would also be great it the GIMP edits could make their way into the textures, so that animations can be exported without effort in the future.

I also just noticed that the carriers don't have player color - maybe make the dresses predominately player color, with colorful fringes/sleeves?

Noordfrees commented:
I don't see how the edits could be incorporated there since these images are rendered from GIMP. I'll add a note how to post-process them though.

The carriers do have player colour but it's not very emphasized. I first tried to make the dresses playercoloured with pc-independent ornaments but this looked really bad. So instead the ornaments on the dresses are made mainly of pc. However I toned this down a bit because any more of it looks just messy.

Noordfrees committed:
Created wilderness keeper

Niektory committed:
Fix huge soldier healing in the field (#3797)

Healing was higher than intended because of min() and max() mix-up.

Noordfrees committed:
Window titlebar buttons (#3763)

Add buttons to close, pin/unpin and minimize/restore windows to every window's titlebar.
Pinned windows cannot be closed by right-click, Esc, or the close button, and cannot be minimized.

rikardfalkeborn committed:
Prettier stdout files when regression_test is run with python 3 (#3800)

Print regression test suite output as raw byte strings

gunchleoc committed:
Animation uncluttering (#3780)

- Make savegames robust against changes in animation names
- Get rid of Lua tables for "walkload" animation if it's identical
  to the "walk" animation
- Prefer "idle" as main animation
- Omit basename when it's identical to the animation name
- Add option to `utils/optimize_pngs.py` for restricting to filename prefix
- Shift definition of animation directory from individual animations to the
  map object in order to declutter the Lua tables.
- Add mipmaps to Barbarian Carrier to test the changes

pospec4444 committed:
Warfare tutorial - Teach the player about scouting (#3774)

* New objective "Build Scout's Hut"
* New objective "Explore the map"

Noordfrees committed:
Added missing `break` in window.cc

Noordfrees committed:
Completed builder

Noordfrees committed:
Completed shipwright

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into amazons

Noordfrees committed:
Merge branch 'amazons' of github.com:widelands/widelands into amazons

Noordfrees committed:
Change amazons.lua to new format

gunchleoc committed:
Add Rikard Falkeborn to developers

gunchleoc committed:
Fix stalling test suite scenarios (#3784)

* `stable_save` now uses last save time and average frame rate for waiting
* pass desired speed explicitly to `stable_save`

tppq committed:
Simplifying the Fellowships map, for better playing experience. (#3792)

This modifies the "Fellowships" map.

Expanding to the southern island requires many small military buildings, and one middle sized one. One has to know which location requires the middle sized building. A casual player first makes a small building, dismantles it and build the larger one. This is annoying. Playing should be pleasant not annoying. Requiring the middle sized one was a bad design choice.

This modifies the map slightly so, all buildings can be small while expanding to south island. No other modifications, unless I was careless with mouse.

Noordfrees committed:
geologist

pospec4444 committed:
More seafaring tutorial improvements (#3788)

- Run waterways tutorial along with shipyard objective
- New objective: Build an iron mine
- Remind player to stop ferry yard production
- Remind player to stop shipyard production when second ship is built
- Evaluate gold mine and ferries simultaneously

Noordfrees committed:
Jungle preserver. Converted completed worker anims to new anim format.

gunchleoc committed:
Add lots of optional braces (#3802)

Noordfrees committed:
Difficulty stages for scenarios (#3795)

Allow campaign designers to define various difficulty stages for campaigns

Noordfrees committed:
Completed jungle master. Fixed walkload pc masks for carrier charcoal_burner cook dressmaker golddigger goldsmelter scout stonecarver tapirbreeder trainer.

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into amazons

jmoerschbach committed:
Refactor saveloading ui (#3748)

- Further separating of concerns: Extracted savegame loading code (SavegameLoader) and savegame deleting code (SavegameDeleter) into own classes
- Introduced small inheritance hierachy for both of these classes to adhere to different loading/deleting of replays/multiplayer/singleplayer
- Show directory names in game details

Noordfrees committed:
Tweaked builtin economy profiles (#3787)

Co-authored-by: WorldSavior

Noordfrees committed:
Liana cutter

gunchleoc committed:
Fetched translations and updated catalogs.

Noordfrees committed:
Woodcutter and master woodcutter

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into amazons

Noordfrees committed:
Stonecutter

Noordfrees committed:
Hunter-gatherer

Noordfrees committed:
Refactor road/waterway building (#3793)

Refactored road and waterway building mode and Lua road building functions and fixed a bug in secret village

Co-authored-by: gunchleoc <fios@foramnagaidhlig.net>

Noordfrees committed:
Re-add quick navigation with comma and period keys (#3799)

Noordfrees committed:
Delete Ctrl-F10 hotkey (#3813)

gunchleoc committed:
Unify program parsers (#3805)

Refactor program parsers
- Pull out common code into new class MapObjectProgram
- Fix ware demand checks so that they only affect the correct tribes
- Get rid of extraneous calls to EditorGameBase::postload()
- Small performance tweak for trainingsite programs
- Get rid of now empty helper library

Noordfrees committed:
Cassavar and Cocoa farmers

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into amazons

gunchleoc committed:
Create CONTRIBUTING.md

Noordfrees committed:
Add missing entries do buildcat.py (#3814)

gunchleoc committed:
Fetched translations and updated catalogs.

pospec4444 committed:
Replace deprecated `message_box_objective()` function (#3807)

- Replaced deprecated function `message_box_objective()`
- "Scolded" message box was too high
- Removed height properties
- Updated image links
- Move new_objectives over from richtext_scenarios

Noordfrees committed:
Ferry

Noordfrees committed:
Fix rookie creation for hero demands (#3804)

Requests for heroes should not trigger rookie production

Noordfrees committed:
Replace deprecated reveal_scenario with mark_scenario_as_solved in frisian campaign (#3825)

* Update fri01

* Update fri02

Noordfrees committed:
Include all scripts in scenarios from the init.lua (#3828)

Noordfrees committed:
Fix a desync (#3824)

Fix desyncs due to an uninitialized variable by exposing the Seed radius random factor of trees to the init.lua

Noordfrees committed:
Immovable program documentation (#3829)

Noordfrees committed:
Ship

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into amazons

Noordfrees committed:
Ship under construction

rikardfalkeborn committed:
Fix off-by-one in get_road() (#3806)

Use `WalkingDir::FIRST_DIRECTION` and `WalkingDir::LAST_DIRECTION` or ranged `for` loops for iterating the roads of a flag

Noordfrees committed:
Amazon resis

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into amazons

actions-user committed:
2 files were automatically formatted.

gunchleoc committed:
Escape richtext in editor info tool (#3798)

Fixes broken info tool texts when the map description or author contains '<' or '>'

jmoerschbach committed:
Suggested teams truncated (#3839)

Set suggested teams box width to correct calculated parent width and remove superfluous `set_size()`

Noordfrees committed:
Amazon rare trees

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into amazons

Noordfrees committed:
Merge branch 'amazons' of github.com:Noordfrees/widelands into amazons

Niektory committed:
Fix road buttons added instead of waterway buttons (#3841)

Fix a bug causing road buttons instead of waterway buttons to be added in waterway building mode when clicking on a field where the waterway can't be built.

gunchleoc committed:
Make wares_in_warehouses the default tab for stock statistics (#3765)

Niektory committed:
Add an option to use SDL mouse cursor (#3810)

Niektory committed:
Add numpad hotkeys (#3840)

* Handle zooming with numpad +/-/0 keys and = key
* Handle numpad keys in the editor
* Handle filtering game messages with numpad keys
* Handle filtering seafaring statistics with numpad keys
* Handle quick navigation with numpad keys

Noordfrees committed:
Fetched translations and updated catalogs.

tppq committed:
2940 better trainingsite soldier selection (#3835)

Trainingsite attempts to get soldiers that already are partially trained. If the trainingsite fails to train, it switches strategy and requests soldiers that are below the failing level. If training succeeds, the site starts requesting strong soldiers again. If the site does not get new soldiers, it will gradually start accepting all trainable soldiers (old way).

gunchleoc committed:
Spritesheets & MipMaps for Barbarian workers (#3779)

Niektory committed:
Fix visual glitches and division by zero in WuiPlotArea (#3838)

Sometimes the plot was drawn without updating highest_scale_ first.
This resulted in the graph being drawn with an incorrect height, possibly outside the statistics window.
And if highest_scale_ was zero, it could also lead to division by zero in scale_value().

Noordfrees committed:
Update CMakeLists for Ubuntu Focal

Noordfrees committed:
Replace dependency python with python3

EXPERIMENTAL: try to fix the PPA for Focal which no longer supports python2.
This commit may need to be undone if it turns out that it breaks support for older distros.

Noordfrees committed:
Try to fix packaging for Focal

Might also fix issue #3783

Noordfrees committed:
Fix travis MacOS builds (#3848)

On Travis MacOS, gettext needs to be added to the PATH and some CMake CXX and linker flags have to be set because brew force-linking is no longer allowed for whatever reason

Noordfrees committed:
Added falling anim for rare trees

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into amazons

hessenfarmer commented:
Don't have time to test this. However as the complete Amazons branch will be reviewed again once ready we could merge this I believe. to clean up our PR section
